### PR TITLE
gdb reg caching

### PR DESCRIPTION
**Pull request by SrimantaBarua**
Small fix to [#7807](https://github.com/radare/radare2/pull/7807), and reg caching in gdb (invalidated upon step, continue, attach/detach and reg_write)

### CODE REVIEW

File: shlr/gdb/src/gdbclient/core.c [hunk 1]
```diff
@@ -139,6 +145,14 @@ int gdbr_connect(libgdbr_t *g, const char *host, int port) {
 	if (strncmp (g->data, "OK", 2)) {
 		// return -1;
 	}
+	// Initialize reg cache
+	reg_cache.maxlen = g->data_max;
```
Comment (radare) [inline shlr/gdb/src/gdbclient/core.c]: maybe check if maxlen < 1?

**Comment by radare** [comment]
can you add an io command to invalidate the reg cache by hand too?

**Comment by SrimantaBarua** [comment]
Okay, under which character are io commands? I can't seem to find anything like this

**Comment by radare** [comment]
!= or \

just implement the system callback in the io plugin and you can run that from there

**Comment by radare** [comment]
Ping. Do you need some help?

**Comment by SrimantaBarua** [comment]
Probably. `!=`  leads [here](https://github.com/radare/radare2/blob/master/libr/core/cmd.c#L1146), so I'm trying to figure out how cons works. Is that the right place?

**Comment by radare** [comment]
please rebase

**Pull request merged**